### ISSUE TITLE
False-positive `Property is never read` for value objects in persisted fields

### ISSUE THREAD

**Issue by fluffycondor** [phpstan/phpstan-doctrine#228]
phpstan-doctrine 1.0.1

The [PHPStan's documentation](https://phpstan.org/blog/detecting-unused-private-properties-methods-constants#what-if-my-code-is-%E2%80%9Cspecial%E2%80%9D%3F) says

> For Doctrine the logic is following:
> Property is always read (it doesn’t need a getter) if it’s a persisted field or association. 

But it doesn't work for value objects used like this:

````php
/**
 * @ORM\Entity(repositoryClass=FooEntityRepository::class)
 */
class FooEntity
{
    /**
     * @ORM\Id
     * @ORM\Column(type="uuid", unique=true)
     */
    private UuidInterface $id;

    /**
     * @ORM\Embedded(class=Bar::class, columnPrefix=false)
     */
    private Bar $bar;
}

/** @ORM\Embeddable */
class Bar
{
    /**
     * @ORM\Column(type="string", length=255)
     */
    private string $bar;
}
````

It leads to `Property Bar::$bar is never read, only written.`

**Comment by ondrejmirtes** [comment]
Yeah, the extension needs to modified for Embeddables it seems.

**Comment by oksidisko** [comment]
Can the fix be expected soon?
